raiden/tests/unit/test_sqlite.py

```diff
@@ -0,0 +1,171 @@
+        expiration=expiration,
+        secret=sha3(factories.make_secret(next(counter))),
+        identifier=next(counter),
+        nonce=1,
+        transferred_amount=0,
+        locked_amount=None,
+        pkey=factories.HOP1_KEY,
+        sender=factories.HOP1,
+    )
+    return from_route, from_transfer
+
+
+def test_get_latest_state_change_by_data_field():
+    """ All state changes which contain a balance proof must be found by when
+    querying the database.
+    """
+    serializer = JSONSerializer
+    storage = SQLiteStorage(':memory:', serializer)
+    counter = itertools.count()
+
+    lock_expired = ReceiveLockExpired(
+        balance_proof=make_signed_balance_proof_from_counter(counter),
+        secrethash=sha3(factories.make_secret(next(counter))),
+        message_identifier=next(counter),
+    )
+    transfer_direct = ReceiveTransferDirect(
+        token_network_identifier=factories.make_address(),
+        message_identifier=next(counter),
+        payment_identifier=next(counter),
+        balance_proof=make_signed_balance_proof_from_counter(counter),
+    )
+    unlock = ReceiveUnlock(
+        message_identifier=next(counter),
+        secret=sha3(factories.make_secret(next(counter))),
+        balance_proof=make_signed_balance_proof_from_counter(counter),
+    )
+    transfer_refund = ReceiveTransferRefund(
+        transfer=make_signed_transfer_from_counter(counter),
+        routes=list(),
+    )
+    transfer_refund_cancel_route = ReceiveTransferRefundCancelRoute(
+        routes=list(),
+        transfer=make_signed_transfer_from_counter(counter),
+        secret=sha3(factories.make_secret(next(counter))),
+    )
+    mediator_from_route, mediator_signed_transfer = make_from_route_from_counter(counter)
+    action_init_mediator = ActionInitMediator(
+        routes=list(),
+        from_route=mediator_from_route,
+        from_transfer=mediator_signed_transfer,
+    )
+    target_from_route, target_signed_transfer = make_from_route_from_counter(counter)
+    action_init_target = ActionInitTarget(
+        route=target_from_route,
+        transfer=target_signed_transfer,
+    )
+
+    statechange_balanceproof = [
+        (lock_expired, lock_expired.balance_proof),
+        (transfer_direct, transfer_direct.balance_proof),
+        (unlock, unlock.balance_proof),
+        (transfer_refund, transfer_refund.transfer.balance_proof),
+        (transfer_refund_cancel_route, transfer_refund_cancel_route.transfer.balance_proof),
+        (action_init_mediator, action_init_mediator.from_transfer.balance_proof),
+        (action_init_target, action_init_target.transfer.balance_proof),
+    ]
+
+    timestamp = datetime.utcnow().isoformat(timespec='milliseconds')
+
+    for state_change, _ in statechange_balanceproof:
+        storage.write_state_change(state_change, timestamp)
+
+    for state_change, balance_proof in statechange_balanceproof:
+        result = storage.get_latest_state_change_by_data_field({
```

Comment:
Note; This is brittle, since it is a copy of the arguments used in the event handler and changes there could not be made here, making the test useless. Suggestions are also welcome.

Reply:
This is now fixed, I'm using the same function for all event handlers and testing it here.